packages/dashboard/src/components/sidePanel/utils/index.ts

```diff
@@ -5,6 +5,11 @@ import { DashboardState } from '~/store/state';
 import { onUpdateWidgetsAction } from '~/store/actions';
 import { AppKitWidget, TextWidget, InputWidget, Widget } from '~/types';
 
+// // TODO: this should work for all widgets so we can access nested paths e.g. 'properties.options'
```

Comment:
i wonder if a better way to approach this is to allow a function which selects what it wants to alter, instead of a path.

Reply:
I think you are right. We'd have to refactor `useInput` to not use `lodash.get`